cube/src/main/java/zone/cogni/asquare/cube/spel/SpelService.java

```diff
@@ -12,23 +12,11 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public class SpelService {
+public class SpelService implements TemplateService {
 
   private final SpelExpressionParser spelExpressionParser = new SpelExpressionParser();
   private final Map<String, Expression> expressionCache = new HashMap<>();
 
-  public NamedTemplate processTemplate(NamedTemplate namedTemplate, Object root) {
-    String result = processTemplate(namedTemplate.getTemplate(), root);
-    return getNewNamedTemplate(namedTemplate, root, result);
-  }
-
-  private NamedTemplate getNewNamedTemplate(NamedTemplate namedTemplate, Object root, String result) {
-    NamedTemplate copy = namedTemplate.copy();
-    copy.setRoot(root);
-    copy.setResult(result);
-    return copy;
-  }
-
   public String processTemplate(Resource templateResource, Object root) {
     String template = ResourceHelper.toString(templateResource);
     return processTemplate(template, root);
```

Comment:
looks like this could be also a default method on the interface (unless there is a special reason for not doing it)